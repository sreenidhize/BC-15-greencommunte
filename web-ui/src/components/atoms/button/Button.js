import React from "react";
import { Button as MuiButton, makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  medium: {
    width: "138px",
    height: "50px",
    padding: theme.spacing(2, 4),
  },
  small: {
    width: "101px",
    height: "38px",
    padding: theme.spacing(1, 4),
  },
  medium2: {
    width: "151px",
    height: "50px",
    padding: theme.spacing(1, 4),
  },
  long: {
    width: "208px",
    height: "50px",
    padding: theme.spacing(1, 4),
  },
  common: {
    borderRadius: "10px",
    fontWeight: "bold",
    fontSize: "14px",
    textTransform: "none",
    fontFamily: theme.typography.fontFamily,
  },
}));

const Button = (props) => {
  const classes = useStyles();
  return (
    <MuiButton
      variant={props.variant}
      color={props.color}
      className={`${classes[props.type]} ${classes.common}`}
      onClick={props.onClick}
      data-testid="button"
    >
      {props.label}
    </MuiButton>
  );
};

Button.defaultProps = {
  variant: "outlined",
  type: "medium",
  label: "button",
  color: "primary",
};

export default Button;
