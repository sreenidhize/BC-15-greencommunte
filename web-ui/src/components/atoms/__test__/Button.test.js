import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../button/Button";


describe("Button test", () => {
    it("Matches the snapshot", () => {
      const { container } = render(
        <Button type="small" label="Test small" variant="contained" color="primary"/>
      );
      expect(container).toMatchSnapshot();
    });
  
    test("Test Button with type small and contained render", () => {
      const { getByTestId } = render(<Button type="small" label="Test small" variant="contained" color="primary"/>);
      expect(getByTestId("button")).toBeInTheDocument();
    });
  
    test("Test Button with type medium and outlined render", () => {
      const { getByTestId } = render(<Button type="medium" label="Test medium" variant="outlined" color="primary"/>);
      expect(getByTestId("button")).toBeInTheDocument();
    });
    test("Test Button with type long and contained render", () => {
        const { getByTestId } = render(<Button type="medium" label="Test medium" variant="contained" color="primary"/>);
        expect(getByTestId("button")).toBeInTheDocument();
      });
  });