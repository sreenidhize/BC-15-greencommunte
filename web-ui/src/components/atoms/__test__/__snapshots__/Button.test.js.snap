// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Matches the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-small-2 makeStyles-common-5 MuiButton-containedPrimary"
    data-testid="button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test small
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
